### ISSUE TITLE
Getting type error on ...createBeautifulMentionNode on lexical latest 0.12.5

### ISSUE THREAD

**Issue by tecoad** [sodenn/lexical-beautiful-mentions#266]
I started getting the following type error after upgrading Lexical:
here:
...createBeautifulMentionNode(CustomMentionComponent)

```
Type '{ new (trigger: string, value: string, data?: { [p: string]: BeautifulMentionsItemData; } | undefined, key?: string | undefined): { [x: string]: any; exportJSON(): SerializedBeautifulMentionNode; ... 54 more ...; markDirty(): void; }; ... 4 more ...; transform(): ((node: LexicalNode) => void) | null; } | { ...; }' is not assignable to type 'Klass<LexicalNode> | LexicalNodeReplacement'.
  Type '{ new (trigger: string, value: string, data?: { [p: string]: BeautifulMentionsItemData; } | undefined, key?: string | undefined): { [x: string]: any; exportJSON(): SerializedBeautifulMentionNode; ... 54 more ...; markDirty(): void; }; ... 4 more ...; transform(): ((node: LexicalNode) => void) | null; }' is not assignable to type 'Klass<LexicalNode> | LexicalNodeReplacement'.
    Type '{ new (trigger: string, value: string, data?: { [p: string]: BeautifulMentionsItemData; } | undefined, key?: string | undefined): { [x: string]: any; exportJSON(): SerializedBeautifulMentionNode; ... 54 more ...; markDirty(): void; }; ... 4 more ...; transform(): ((node: LexicalNode) => void) | null; }' is not assignable to type 'Klass<LexicalNode>'.
      Type '{ new (trigger: string, value: string, data?: { [p: string]: BeautifulMentionsItemData; } | undefined, key?: string | undefined): { [x: string]: any; exportJSON(): SerializedBeautifulMentionNode; ... 54 more ...; markDirty(): void; }; ... 4 more ...; transform(): ((node: LexicalNode) => void) | null; }' is not assignable to type 'new (...args: any[]) => LexicalNode'.
        Type '{ [x: string]: any; exportJSON(): SerializedBeautifulMentionNode; component(): ElementType<BeautifulMentionComponentProps<{}>> | null; ... 53 more ...; markDirty(): void; }' is missing the following properties from type 'LexicalNode': selectStart, selectEndts(2322)
```

The error message indicates that the return type of createBeautifulMentionNode(CustomMentionComponent) is not assignable to the type Klass<LexicalNode> | LexicalNodeReplacement. This means that the object returned by createBeautifulMentionNode(CustomMentionComponent) does not match the expected type.

**Comment by sodenn** [comment]
Hi @tecoad. Thanks for reporting. They definitely changed some types in 0.12.5, but I didn't see this error after upgrading the demo app to [0.12.5](https://github.com/sodenn/lexical-beautiful-mentions/commit/1199d8832cae53e898760641bed7bb3b0d90ed22). You can find the editor config of the demo app [here](https://github.com/sodenn/lexical-beautiful-mentions/blob/main/www/lib/editor-config.ts). Do you see any differences between your code and the config in the demo app that could cause this error? Is it only a type error or does it also break the editor?

**Comment by GlennOlsson** [comment]
For me, it was only a type error, it did not break the runtime. Solved it by downgrading as @tecoad suggested

**Comment by sodenn** [comment]
I've added some missing types to `BeautifulMentionNode` to fix https://github.com/sodenn/lexical-beautiful-mentions/issues/321 and released a new version (v0.1.28).
I think this could also be the reason for this error. Can you please check if the error is gone now?
